### ISSUE TITLE
Characters parsed into unicode when using Decode_Field_As: escaped for the docker parser

### ISSUE THREAD

**Issue by gabe-sorensen** [fluent/fluent-bit#580]
I'm was working with the 0.13-dev:0.18 image, and was running into a few issues with the parsing, namely the handling of the characters '<', '>', and '&'. I'm not sure if there's other cases, since that's all I've seen in my logs. Those characters get parsed to 'u003c', 'u003e', and 'u0026', respectively in the output. I've also tested with 0.13.0, and have seen the same issue.

Old version (before switching to 0.13-dev:0.18. Notice \n is not removed)
2018-04-23 20:33:37,656 [TEST] [   DEBUG] [urllib3.util.retry] [pid:17   ] [    MainThread] Converted **retries value: 3 -> Retry(total=3**, connect=None, read=None, redirect=None, status=None)\n

New version (with 0.13-dev:0.18. \n is correctly removed, but the > character has been modified)
2018-04-23 20:33:37,656 [TEST] [   DEBUG] [urllib3.util.retry] [pid:17   ] [    MainThread] Converted **retries value: 3 -u003e Retry(total=3**, connect=None, read=None, redirect=None, status=None)

Old "correct" record:
```json
{
  "_index": "logstash-2018.04.23",
  "_type": "fluentd",
  "_id": "AWL0NlCOJkRsMcLZl-SA",
  "_score": null,
  "_source": {
    "log": "2018-04-23 20:33:37,656 [TEST] [   DEBUG] [urllib3.util.retry] [pid:17   ] [    MainThread] Converted retries value: 3 -> Retry(total=3, connect=None, read=None, redirect=None, status=None)\n",
    "stream": "stderr",
    "docker": {
      "container_id": "92ca99fcc91bd460739e930a4e6f87139fa2c68042f2760f411aece98a778c2b"
    },
    "kubernetes": {
      "container_name": "test-container",
      "namespace_name": "qa1-ns",
      "pod_name": "test-webserver-847755c865-rz8x5",
      "pod_id": "fef4a613-4732-11e8-8cbb-06712f6f161e",
      "labels": {
        "app": "test-webserver",
        "pod-template-hash": "4033117421",
        "release": "qa1-test"
      },
      "host": "ip-10-100-22-172.us-west-2.compute.internal",
      "master_url": "https://100.64.0.1:443/api",
      "namespace_id": "cb57e0e3-38e1-11e8-9d82-06e3398080f4"
    },
    "@timestamp": "2018-04-23T20:33:37+00:00",
    "tag": "kubernetes.var.log.containers.test-webserver-847755c865-rz8x5_qa1-ns_test-container-92ca99fcc91bd460739e930a4e6f87139fa2c68042f2760f411aece98a778c2b.log"
  },
  "fields": {
    "@timestamp": [
      1524515617000
    ]
  },
  "highlight": {
    "log": [
      "2018-04-23 20:33:37,656 [TEST] [   DEBUG] [urllib3.util.retry] [pid:17   ] [    MainThread] @kibana-highlighted-field@Converted@/kibana-highlighted-field@ @kibana-highlighted-field@retries@/kibana-highlighted-field@ @kibana-highlighted-field@value@/kibana-highlighted-field@: 3 -> Retry(total=3, connect=None, read=None, redirect=None, status=None)\n"
    ],
    "kubernetes.labels.release": [
      "@kibana-highlighted-field@qa1@/kibana-highlighted-field@-@kibana-highlighted-field@test@/kibana-highlighted-field@"
    ]
  },
  "sort": [
    1524515617000
  ]
}
```

New record: 
```json
{
  "_index": "logstash-2018.04.23",
  "_type": "flb_type",
  "_id": "AWL0P8cFIevz06KnfN3W",
  "_score": null,
  "_source": {
    "@timestamp": "2018-04-23T20:33:37.656Z",
    "log": "2018-04-23 20:33:37,656 [TEST] [   DEBUG] [urllib3.util.retry] [pid:17   ] [    MainThread] Converted retries value: 3 -u003e Retry(total=3, connect=None, read=None, redirect=None, status=None)",
    "stream": "stderr",
    "time": "2018-04-23T20:33:37.656915306Z",
    "kubernetes": {
      "pod_name": "test-webserver-847755c865-rz8x5",
      "namespace_name": "qa1-ns",
      "pod_id": "fef4a613-4732-11e8-8cbb-06712f6f161e",
      "labels": {
        "app": "test-webserver",
        "pod-template-hash": "4033117421",
        "release": "qa1-test"
      },
      "annotations": {
        "iam_amazonaws_com/role": "test",
        "kubernetes_io/created-by": "{\\\"kind\\\":\\\"SerializedReference\\\",\\\"apiVersion\\\":\\\"v1\\\",\\\"reference\\\":{\\\"kind\\\":\\\"ReplicaSet\\\",\\\"namespace\\\":\\\"qa1-ns\\\",\\\"name\\\":\\\"test-webserver-847755c865\\\",\\\"uid\\\":\\\"64d461b0-449a-11e8-9d82-06e3398080f4\\\",\\\"apiVersion\\\":\\\"extensions\\\",\\\"resourceVersion\\\":\\\"9114439\\\"}}\\n",
        "kubernetes_io/limit-ranger": "LimitRanger plugin set: cpu, memory limit for container test-container"
      },
      "host": "ip-10-100-22-172.us-west-2.compute.internal",
      "container_name": "test-container",
      "docker_id": "92ca99fcc91bd460739e930a4e6f87139fa2c68042f2760f411aece98a778c2b"
    }
  },
  "fields": {
    "@timestamp": [
      1524515617656
    ],
    "time": [
      1524515617656
    ]
  },
  "sort": [
    1524515617656
  ]
}
```
Configmap:
```yaml
apiVersion: v1
kind: ConfigMap
metadata:
  name: fluent-bit-config
  namespace: logging
  labels:
    k8s-app: fluent-bit
data:
  # Configuration files: server, input, filters and output
  # ======================================================
  fluent-bit.conf: |
    [SERVICE]
        Flush         1
        Log_Level     info
        Daemon        off
        Parsers_File  parsers.conf
        HTTP_Server   On
        HTTP_Listen   0.0.0.0
        HTTP_Port     2020

    @INCLUDE input-kubernetes.conf
    @INCLUDE filter-kubernetes.conf
    @INCLUDE output-elasticsearch.conf

  input-kubernetes.conf: |
    [INPUT]
        Name              tail
        Tag               kube.*
        Path              /var/log/containers/*.log
        Parser            docker
        DB                /var/log/flb_kube.db
        Mem_Buf_Limit     5MB
        Skip_Long_Lines   On
        Refresh_Interval  10

  filter-kubernetes.conf: |
    [FILTER]
        Name                kubernetes
        Match               kube.*
        Kube_URL            https://kubernetes.default.svc:443
        Merge_Log           On
        K8S-Logging.Parser  On

  output-elasticsearch.conf: |
    [OUTPUT]
        Name            es
        Match           *
        Host            ${FLUENT_ELASTICSEARCH_HOST}
        Port            ${FLUENT_ELASTICSEARCH_PORT}
        Logstash_Format On
        Logstash_Prefix logstash
        TLS             On
        Retry_Limit     False

  parsers.conf: |
    [PARSER]
        Name   apache
        Format regex
        Regex  ^(?<host>[^ ]*) [^ ]* (?<user>[^ ]*) \[(?<time>[^\]]*)\] "(?<method>\S+)(?: +(?<path>[^\"]*?)(?: +\S*)?)?" (?<code>[^ ]*) (?<size>[^ ]*)(?: "(?<referer>[^\"]*)" "(?<agent>[^\"]*)")?$
        Time_Key time
        Time_Format %d/%b/%Y:%H:%M:%S %z

    [PARSER]
        Name   apache2
        Format regex
        Regex  ^(?<host>[^ ]*) [^ ]* (?<user>[^ ]*) \[(?<time>[^\]]*)\] "(?<method>\S+)(?: +(?<path>[^ ]*) +\S*)?" (?<code>[^ ]*) (?<size>[^ ]*)(?: "(?<referer>[^\"]*)" "(?<agent>[^\"]*)")?$
        Time_Key time
        Time_Format %d/%b/%Y:%H:%M:%S %z

    [PARSER]
        Name   apache_error
        Format regex
        Regex  ^\[[^ ]* (?<time>[^\]]*)\] \[(?<level>[^\]]*)\](?: \[pid (?<pid>[^\]]*)\])?( \[client (?<client>[^\]]*)\])? (?<message>.*)$

    [PARSER]
        Name   nginx
        Format regex
        Regex ^(?<remote>[^ ]*) (?<host>[^ ]*) (?<user>[^ ]*) \[(?<time>[^\]]*)\] "(?<method>\S+)(?: +(?<path>[^\"]*?)(?: +\S*)?)?" (?<code>[^ ]*) (?<size>[^ ]*)(?: "(?<referer>[^\"]*)" "(?<agent>[^\"]*)")?$
        Time_Key time
        Time_Format %d/%b/%Y:%H:%M:%S %z

    [PARSER]
        Name   json
        Format json
        Time_Key time
        Time_Format %d/%b/%Y:%H:%M:%S %z

    [PARSER]
        Name        docker
        Format      json
        Time_Key    time
        Time_Format %Y-%m-%dT%H:%M:%S.%L
        Time_Keep   On
        # Command      |  Decoder | Field | Optional Action
        # =============|==================|=================
        Decode_Field_As   escaped    log

    [PARSER]
        Name        syslog
        Format      regex
        Regex       ^\<(?<pri>[0-9]+)\>(?<time>[^ ]* {1,2}[^ ]* [^ ]*) (?<host>[^ ]*) (?<ident>[a-zA-Z0-9_\/\.\-]*)(?:\[(?<pid>[0-9]+)\])?(?:[^\:]*\:)? *(?<message>.*)$
        Time_Key    time
        Time_Format %b %d %H:%M:%S
```

daemonset:
```yaml
apiVersion: extensions/v1beta1
kind: DaemonSet
metadata:
  name: fluent-bit
  namespace: logging
  labels:
    k8s-app: fluent-bit-logging
    version: v1
    kubernetes.io/cluster-service: "true"
spec:
  template:
    metadata:
      labels:
        k8s-app: fluent-bit-logging
        version: v1
        kubernetes.io/cluster-service: "true"
      annotations:
        prometheus.io/scrape: "true"
        prometheus.io/port: "2020"
        prometheus.io/path: /api/v1/metrics/prometheus
    spec:
      containers:
      - name: fluent-bit
        image: fluent/fluent-bit-0.13-dev:0.18
        imagePullPolicy: Always
        ports:
          - containerPort: 2020
        env:
        - name: FLUENT_ELASTICSEARCH_HOST
          value: "elasticsearch_url.example.com"
        - name: FLUENT_ELASTICSEARCH_PORT
          value: "443"
        volumeMounts:
        - name: varlog
          mountPath: /var/log
        - name: varlibdockercontainers
          mountPath: /var/lib/docker/containers
          readOnly: true
        - name: fluent-bit-config
          mountPath: /fluent-bit/etc/
      terminationGracePeriodSeconds: 10
      volumes:
      - name: varlog
        hostPath:
          path: /var/log
      - name: varlibdockercontainers
        hostPath:
          path: /var/lib/docker/containers
      - name: fluent-bit-config
        configMap:
          name: fluent-bit-config
      serviceAccountName: fluent-bit
      tolerations:
      - key: node-role.kubernetes.io/master
        operator: Exists
        effect: NoSchedule
```

**Comment by ffscl** [comment]
@gabe-sorensen Try the decoder implemented on this PR: #568

**Comment by jcassee** [comment]
I had this problem too, and the `escaped_utf8` decoder fixed it for me.

**Comment by gabe-sorensen** [comment]
Looks like using that decoder will work. Thanks!

**Comment by edsiper** [comment]
FYI:

Please check the following comment on #1278 : 

https://github.com/fluent/fluent-bit/issues/1278#issuecomment-499583503

**Comment by edsiper** [comment]
Issue already fixed, ref: #1278 (comment)
